### ISSUE TITLE
Default parameters in interop setters are allowed

### ISSUE THREAD

**Issue by gzm0** [scala-js/scala-js#1707]
The two following definitions compile, but probably shouldn't:

```scala
trait JS extends js.Object {
  def a_=(x: Int = 1): Unit = js.native
}

class A {
  @JSExport
  def a_=(x: Int = 1): Unit = ???
}
```

**Comment by gzm0** [comment]
@sjrd, please confirm this.

**Comment by sjrd** [comment]
The raw JS type variant should definitely not compile (the compiler wouldn't know what to do with a call site `x.a_=()`!)

For the exports, well ... Tough question. The default would bear no meaning for the export, but could in theory be used by Scala code. I would say forbid it. (If this is really what the author wants, they can declare an overload that is not exported.)

**Comment by gzm0** [comment]
> I would say forbid it.

I agree. It is something that should not occur in Scala code either. And the overload (or a second method) is a reasonable workaround if one really needs it.

**Comment by gzm0** [comment]
I assume we'll emit a warning for this in the 0.6.x series, right?

**Comment by sjrd** [comment]
Yep, I guess we have to make it a warning.

We should also emit a compile error for a call site `x.a_=()`. (Currently it would likely silently produce invalid IR with a leftover `UndefinedParam()` or something like that.)

**Comment by sjrd** [comment]
Fixed in 7cacc307c3a9184ce317268f9489353bc8a862a4
